streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -623,16 +613,46 @@ void runOnce() {
             return;
         }
 
-        initializeAndRestorePhase();
+        // we need to first add closed tasks and then created tasks to work with those revived / recycled tasks
+        restoreThread.addClosedTasks(taskManager.drainRemovedTasks());
+
+        // try to initialize created tasks that are either newly assigned or re-created from corrupted tasks
+        final List<AbstractTask> initializedTasks;
+        if (!(initializedTasks = taskManager.tryInitializeNewTasks()).isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Initializing newly created tasks {} under state {}",
+                        initializedTasks.stream().map(AbstractTask::id).collect(Collectors.toList()), state);
+            }
+
+            restoreThread.addInitializedTasks(initializedTasks);
+        }
+
+        // try complete restoration if there are any restoring tasks
+        if (taskManager.tryToCompleteRestoration(restoreThread.completedChangelogs())) {
+            log.debug("Completed restoring all tasks now");
+        }
+
+        if (state == State.PARTITIONS_ASSIGNED && taskManager.allTasksRunning()) {
+            // it is possible that we have no assigned tasks in which case we would still transit state
+            setState(State.RUNNING);
+
+            log.debug("All tasks are now running and transited State to {}", State.RUNNING);
```

Comment:
```suggestion
            log.info("All tasks are now running");
```
(the `setState` call will log the state change so we don't really need to log it again here imo)